smart_open/smart_open_lib.py

```diff
@@ -163,6 +164,14 @@ def smart_open(uri, mode="rb", **kw):
     if not isinstance(mode, six.string_types):
         raise TypeError('mode should be a string')
 
+    # Support opening ``pathlib.Path`` objects by casting them to strings.
+    try:
+        from pathlib import Path
```

Comment:
Oh, `pathlib` available only for `python>=3.4`, that's sad

Reply:
Instead of checking for `python>=3.4` and assuming `pathlib` support I opted for the try-except pattern since there is a backport of `pathlib` that users may wish to use for `python<3.4`:

https://pypi.python.org/pypi/pathlib2/

Reply:
Maybe add it to `setup.py`, wdyt @mpenkov?  https://github.com/RaRe-Technologies/smart_open/blob/ecb24c23f4ad58904d371d88eeb1f83389cae16e/setup.py#L19-L23